src/lib/ParamServerApiClient.js

```diff
@@ -88,15 +88,14 @@ class ParamServerApiClient {
     ];
 
     return new Promise((resolve, reject) => {
-      this._xmlrpcClient.methodCall('hasParam', data, (err, resp) => {
-        if (err || resp[0] !== 1) {
-          reject(err, resp);
-        }
-        else {
+      this._call('hasParam', data, (resp) => {
+        if (resp[0] !== 1) {
```

Comment:
I think you can ignore this bit - `XmlrpcClient` should check if the response code is not 1 and automatically reject the call. You should be able to just call `resolve(resp[2])` in this callback

Reply:
OK, please see the second commit!

I don't know where in the code the structure of the response array is decided, but would it make sense to define named constants for the indices like `const ERROR_CODE = 0;` or `const REQUESTED_VALUE = 2;` so you could write `resolve(resp[REQUESTED_VALUE]);` and wouldn't need the comment `// resp[2] is whether it actually has param and presumably all anyone  cares about`?

Reply:
The response array follows the ROS XMLRPC Master/Slave/Parameter API [spec](http://wiki.ros.org/ROS/Master_Slave_APIs). Generally the first element is a status code, the second element is a status message, and the third element is the actual data.

Unfortunately, this was one of the first things I wrote and so the use of different indices is scattered across a few files. We should be able to move all of that into XmlRpcClient and have it deal with any indexing into the actual response, but it would be a bit more work than just adding constants into one file.

It's been towards the back of my "nice to do" list for a while, but you're welcome to dig into it if you want.